### DISCUSSION TITLE
Any tools to extract translations from a spreadsheet like google sheets and append keys to *.po files?

### DISCUSSION THREAD

**Discussion by zxlvera** [lingui/js-lingui#2081] [Q&A]
I am testing out lingui as an alternative to i18next.

Currently our repo extracts from google sheets into a json file to be placed in `locales/` for i18next to read the translation keys.
Any similar tools for lingui where translated text in google sheets will output as *.po files?

**Comment by andrii-bodnar** [comment]
Hi @zxlvera,

While it's technically possible, it's not recommended to use Google Sheets for translation. Consider using a [Translation Management System (TMS)](https://lingui.dev/tools/introduction) to synchronize your Lingui `*.po` files for translation and download the translated files.

The TMS-based workflow would be much simpler and easier in this case. In addition, it brings a lot of tools and capabilities for quality translations. All your existing translations could be imported as translation memory in this case and you will reuse all of them after integrating Lingui.

**Reply by zxlvera** [reply]
thank you, TMS sounds great. Unfortunately my company doesn't want to spend on such services. I would definitely use this on my personal projects

**Reply by andrii-bodnar** [reply]
There are also free options, depending on the size of the project.

By the way, converting PO to spreadsheet and back can be tricky. You might be interested in some of the other [catalog formats](https://lingui.dev/ref/catalog-formats) supported by Lingui that might simplify your current workflow.
